marconi-chain-index/src/Marconi/ChainIndex/Experimental/Indexers/MintTokenEvent.hs

```diff
@@ -354,15 +354,51 @@ mkMintTokenIndexer dbPath = do
                  redeemerData
               ) VALUES
               (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)|]
-      createMintPolicyEventTables =
-        [createMintPolicyEvent, createMintPolicyIdIndex]
+      createMintPolicyEventTables = [createMintPolicyEvent]
       mintInsertPlans = [Core.SQLInsertPlan fromTimedMintEvents mintEventInsertQuery]
   Sync.mkSyncedSqliteIndexer
     dbPath
     createMintPolicyEventTables
     [mintInsertPlans]
     [Core.SQLRollbackPlan "minting_policy_events" "slotNo" C.chainPointToSlotNo]
 
+catchupConfigEventHook :: Trace IO Text -> FilePath -> Core.CatchupEvent -> IO ()
+catchupConfigEventHook stdoutTrace dbPath Core.Synced = do
+  SQL.withConnection dbPath $ \c -> do
+    let txIdPolicyIdIndexName = "minting_policy_events__txId_policyId"
+        createMintPolicyIdIndexStatement =
+          "CREATE INDEX IF NOT EXISTS "
+            <> fromString txIdPolicyIdIndexName
+            <> " ON minting_policy_events (txId, policyId)"
+    createIndexTable c txIdPolicyIdIndexName createMintPolicyIdIndexStatement
+
+    let slotNoIndexName = "minting_policy_events__slotNo"
+        createSlotNoIndexStatement =
+          "CREATE INDEX IF NOT EXISTS "
+            <> fromString slotNoIndexName
+            <> " ON minting_policy_events (slotNo)"
+    createIndexTable c slotNoIndexName createSlotNoIndexStatement
+  where
+    createIndexTable :: SQL.Connection -> String -> SQL.Query -> IO ()
+    createIndexTable c indexName createIndexStatement = do
+      (doesIndexExist :: Bool) <-
+        liftIO $
+          fmap ((> (0 :: Int)) . sum . concat) $
+            SQL.query_
+              c
+              ( [sql|SELECT COUNT(*)
+                  FROM sqlite_master
+                  WHERE type='index' AND name='|]
+                  <> fromString indexName
+                  <> "'"
+              )
+      unless doesIndexExist $ do
+        logInfo stdoutTrace $
+          "Creating SQL index table '" <> Text.pack indexName <> "' for MintTokenEvent indexer..."
+        liftIO $ SQL.execute_ c createIndexStatement
+        logInfo stdoutTrace $
+          "Finished creating SQL index table '" <> Text.pack indexName <> "' for MintTokenEvent indexer..."
+
```

Comment:
As `CREATE INDEX IF NOT EXISTS` already takes care of not doing anything if we have the index in place, do we want to check it manually to provide a feedback?

Reply:
Exactly, it's just for the logging.